File: app/src/main/java/com/webviewblock/navigator/SharedEvents.java
package com.webviewblock.navigator;

import androidx.lifecycle.MutableLiveData;

import com.webviewblock.util.SingleLiveEvent;

import java.io.Serializable;

public class SharedEvents {
    MutableLiveData<NavigationEventData> navigationEvent = new MutableLiveData<NavigationEventData>();
    SingleLiveEvent<Serializable> navigationResultEvent = new SingleLiveEvent<Serializable>();
}

File: app/src/main/java/com/webviewblock/navigator/SharedEvents.kt
package com.webviewblock.navigator

import androidx.lifecycle.MutableLiveData
import com.webviewblock.util.SingleLiveEvent
import java.io.Serializable

open class SharedEvents {
    val navigationEvent = MutableLiveData<NavigationEventData>()
    val navigationResultEvent = SingleLiveEvent<Serializable>()
}